src/main/java/it/gov/pagopa/payment/instrument/service/idpaycode/PaymentInstrumentCodeServiceImpl.java

```diff
@@ -99,6 +111,50 @@ public boolean codeStatus(String userId) {
     return idpayCodeEnabled;
   }
 
+  /** Verify if pin block is correct */
+  @Override
+  public boolean verifyPinBlock(String userId, PinBlockDTO pinBlockDTO) {
+    PaymentInstrumentCode paymentInstrumentCode = paymentInstrumentCodeRepository.findByUserId(
+        userId).orElse(null);
+    if (paymentInstrumentCode == null){
+      throw new PaymentInstrumentException(404, "");
+    }
+
+    String encryptedShaDataBlock = idpayCodeEncryptionService.hashSHADecryptedDataBlock(userId, pinBlockDTO,
```

Comment:
this is not encrypted
```suggestion
    String inputPlainIdpayCode = idpayCodeEncryptionService.hashSHADecryptedDataBlock(userId, pinBlockDTO,
```